docs/docs/50-user-guide/60-reference-docs/40-expressions.md

```diff
@@ -371,9 +371,54 @@ relevant `Freight` is not found from the `FreightCollection`.
 :::tip
 You can handle `nil` values gracefully in Expr using its
 [nil coalescing](https://expr-lang.org/docs/language-definition#nil-coalescing) and
-[optional chaining](https://expr-lang.org/docs/language-definition#optional-chaining) features.
+[optional chaining](https://expr-lang.org/docs/language-definition#optional-chaining)
+features.
 :::
 
+### `freightMetadata(freightName)`
+
+The `freightMetadata()` function retrieves metadata stored in a `Freight` resource.
+It has one required argument:
+
+- `freightName` (Required): The name of the `Freight` resource
+
+This returns a map containing all metadata key-value pairs stored in the `Freight`
+resource.
+
+Example:
+
+```yaml
+config:
+  # Access metadata values using ['key-name'] syntax
+  issueKey: ${{ freightMetadata(ctx.targetFreight.name)['jira-issue-key'] }}
+  customValue: ${{ freightMetadata(ctx.targetFreight.name)['my-custom-key'] }}
+```
+
+:::note
+The two-argument form (`freightMetadata(ctx.targetFreight.name, 'jira-issue-key')`)
+is deprecated as of `v1.8` and will be removed in `v1.10`. Please migrate to the
+single-argument form as shown above.
+:::
```

Comment:
The definite article "the" implies familiarity, making "the two-argument form" sound a little bit as if this is something that was previously discussed, but it wasn't. I would consider rewording things a bit here, to something like "An optional second argument is supported, but is deprecated as of ..." It might also be helpful to, briefly, note how the return values of the two argument and one argument forms differ. It will be helpful to those who do need to migrate.